ballerina/client.bal

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# Milvus is a high-performance, cloud-native vector database built for scalable vector ANN search.
+#
+public isolated client class Client {
+
+    # Gets invoked to initialize the `connector`.
+    #
+    # + config - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(string serviceUrl, *ConnectionConfig config) returns error? {
+        return self.initiateClient(serviceUrl, config);
+    }
+
+    private isolated function initiateClient(string serviceUrl, *ConnectionConfig config) returns error? = @java:Method {
+        'class: "io.ballerina.lib.milvus.Client"
+    } external;
+
+    # Lists all the collections in the Milvus vector database.
+    #
+    # + return - A list of collection names
+    remote isolated function listCollections() returns string[] = @java:Method {
+        'class: "io.ballerina.lib.milvus.Client"
+    } external;
+
+    // TODO: Add support for both dynamic fields and collection schema when creating a collection
+    # Creates a new collection in the Milvus vector database.
+    #
+    # + request - The request to create a collection
+    # + return - `()` if the collection is created successfully, otherwise an error
+    remote isolated function createCollection(CreateCollectionRequest request) returns error? = @java:Method {
```

Comment:
We can create a separate error type for this module if younprefered, not a mandatory thing

Reply:
Fixed in [235867f](https://github.com/ballerina-platform/module-ballerinax-milvus/pull/1/commits/235867faa8998c95845ebf1745789ebf8c032844)